stdlib/InteractiveUtils/test/runtests.jl

```diff
@@ -249,10 +249,10 @@ end
         @test occursin("Environment:", ver)
     end
     let exename = `$(Base.julia_cmd()) --startup-file=no`
-        @test !occursin("Environment:", read(setenv(`$exename -e 'using InteractiveUtils; versioninfo()'`,
-                                                    String[]), String))
-        @test  occursin("Environment:", read(setenv(`$exename -e 'using InteractiveUtils; versioninfo()'`,
-                                                    String["JULIA_CPU_THREADS=1"]), String))
+        @test !occursin("Environment:", read(setenv(`$exename -e 'using InteractiveUtils; versioninfo()'`, String[]), String))
+        ecpy = copy(ENV)
+        ecpy["JULIA_CPU_THREADS"] = "1"
+        @test  occursin("Environment:", read(setenv(`$exename -e 'using InteractiveUtils; versioninfo()'`, ecpy), String))
```

Comment:
Maybe it would be better to pass `--project=$dir` for some `dir` than trying to carry `ENV` fully? I am not sure why `--project` was not included in the default `Base.julia_cmd()` flags (which is roughly doing a `fork`). Maybe @StefanKarpinski or @KristofferC remembers?

Reply:
I don't think there is any particular reason and it should probably be added.

Reply:
A lot of code will need to be rewritten if `Base.julia_cmd()` suddenly starts including `--project`. Right now, if you want to start a new Julia process with a different project, you can simply take `Base.julia_cmd()` and do a `setenv` with `JULIA_PROJECT`. If `Base.julia_cmd()` starts to include `--project`, then the `--project` would override the `JULIA_PROJECT`, which will lead to a lot of people's code breaking in hard-to-diagnose ways.